### ISSUE TITLE
Proposal: dont_return_untyped_future

### ISSUE THREAD

**Issue by bw-flagship** [dart-lang/linter#4362]
**Description**
It is a bad practice to return just the type "Future" instead of "Future<void>", "Future<String>" etc.

**Details**
The compiler is quite good and detects the return value automatically. However, It is more stable, easier to understand and less open to errors if the type is specified.

**Bad Examples**
```dart
Future f() async{
  return "hello world";
}
```

**Good Examples**
```dart
Future<void> f() async{
  return "hello world";
}
```

**Comment by bw-flagship** [comment]
Closed, can be done via

  strong-mode:
    implicit-casts: false
    implicit-dynamic: false

**Comment by eernstg** [comment]
You might want to use `strict-raw-types` in 'analysis_options.yaml' as well, e.g.:

```yaml
analyzer:
  language:
    strict-raw-types: true
    strict-inference: true
    strict-casts: true
```
